File: src/main/java/com/beta_foprhoton/creativecomputerbugs/foundation/helpers/BlockEntityHelpers.java
package com.beta_foprhoton.creativecomputerbugs.foundation.helpers;

import com.beta_foprhoton.creativecomputerbugs.CCBMain;
import com.beta_foprhoton.creativecomputerbugs.foundation.computercraft.core.block.WormComputerHolder;
import net.minecraft.world.level.block.entity.BlockEntity;

import javax.annotation.Nullable;

import static com.beta_foprhoton.creativecomputerbugs.CCBMain.INFECTED_BLOCK_FLAG;

public class BlockEntityHelpers {

    @Nullable
    public static WormComputerHolder getBugComputerHolder(BlockEntity blockEntity) {
        if (blockEntity.getPersistentData().contains(INFECTED_BLOCK_FLAG))
            return CCBMain.BUG_COMPUTER_HOLDER_REGISTER.getWormComputerHolder(blockEntity.getPersistentData().getInt(INFECTED_BLOCK_FLAG));
        return null;
    }
}

File: src/main/kotlin/com/betafoprhoton/creativecomputerbugs/foundation/helpers/BlockEntityHelpers.kt
package com.betafoprhoton.creativecomputerbugs.foundation.helpers;

import com.betafoprhoton.creativecomputerbugs.CCBMain;
import com.betafoprhoton.creativecomputerbugs.CCBMain.Companion.INFECTED_BLOCK_FLAG
import com.betafoprhoton.creativecomputerbugs.foundation.computercraft.core.block.WormComputerHolder;
import net.minecraft.world.level.block.entity.BlockEntity;

import javax.annotation.Nullable;
fun getBugComputerHolder(blockEntity: BlockEntity): WormComputerHolder? {
    if (blockEntity.getPersistentData().contains(INFECTED_BLOCK_FLAG))
        return CCBMain.BUG_COMPUTER_HOLDER_REGISTER.getWormComputerHolder(blockEntity.getPersistentData().getInt(INFECTED_BLOCK_FLAG));
    return null
}
